Tests/AppTests/ArrayVersionExtensionTests.swift

```diff
@@ -0,0 +1,178 @@
+// Copyright Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+@testable import App
+
+import SemanticVersion
+import SPIManifest
+
+class ArrayVersionExtensionTests: AppTestCase {
+
+    func test_Array_canonicalDocumentationTarget() async throws {
+        let pkg = try await savePackageAsync(on: app.db, "1".url)
+        let archive = DocArchive(name: "foo", title: "Foo")
+
+        do {
+            // [ Default branch, nil, nil ] = Default branch
+            let versions = [
+                try Version(package: pkg,
+                            commit: "123",
+                            commitDate: .t0,
+                            docArchives: [archive],
+                            latest: .defaultBranch,
+                            reference: .branch("main"),
+                            spiManifest: nil),
+                nil,
+                nil
+            ]
+
+            XCTAssertEqual(versions.canonicalDocumentationTarget(),
+                           .internal(reference: "main", archive: "foo"))
+        }
+
+        do {
+            // [ Default branch (with docs), Release (no docs), nil ] = Release
```

Comment:
Comment looks wrong, = default branch says the test (I'd drop these comments and let the tests "speak" tbh, that avoids drift)

Reply:
well actually, given how big the Version init is, the comment does help outline the gist of the array input, so it does help